internal/feed/changedetection.go

```diff
@@ -0,0 +1,79 @@
+package feed
+
+import (
+	"fmt"
+	"log/slog"
+	"net/http"
+	"time"
+)
+
+type changeDetectionResponseJson struct {
+	Name        string `json:"title"`
+	Url         string `json:"url"`
+	LastChanged int    `json:"last_changed"`
+}
+
+	
+func parseLastChangeTime(t int) time.Time {
+	parsedTime := time.Unix(int64(t), 0)
+	return parsedTime
+}
+
+
+func FetchLatestDetectedChanges(watches []string, token string) (ChangeWatches, error) {
+	changeWatches := make(ChangeWatches, 0, len(watches))
+
+	if len(watches) == 0 {
+		return changeWatches, nil
+	}
+
+	requests := make([]*http.Request, len(watches))
+
+	for i, repository := range watches {
+		request, _ := http.NewRequest("GET", fmt.Sprintf("https://changedetection.knhash.in/api/v1/watch/%s", repository), nil)
```

Comment:
I think this was meant to be a configurable URL? 😄

Reply:
Ouch, yes, this is meant to be configurable. Will change.